how2meet/ui/components/event_editor.py

```diff
@@ -0,0 +1,136 @@
+from functools import partial
+from datetime import datetime
+import uuid
+
+from nicegui import ui
+
+from how2meet.utils import APIClient as api
+from how2meet.ui.pages.urls import URL_EVENTS, URL_NEW_EVENT, URL_EVENT_HOME
+
+class InviteEditor:
+
+        def __init__(self, invite_id: str):
+            self.invite_id = invite_id
+            self.invite = None
+
+        async def load(self):
+            # self.invite = await api.get_invite(self.invite_id)
+            pass
+
+        async def delete(self):
+            self.card.delete()
+
+        async def render(self):
+            await self.load()
+            # TODO add method to remove guest
+            with ui.card().classes("flex-grow position:relative") as card:
+                self.card = card
+                with ui.row().classes("flex-grow justify-between items-center"):
+                    self.name_input = ui.input("Name")
+                    self.email_input = ui.input("Email")
+                    self.phone_input = ui.input("Phone Number")
+                    ui.button("", icon="delete", color="red", on_click=self.delete).classes("w-6 h-6")
+
+
+        def model_dump(self):
+            return {
+                "id": self.invite_id,
+                "name": self.name_input.value,
+                "email": self.email_input.value,
+                "phone": self.phone_input.value,
+                "status": self.status_input.value,
+                "password": self.password_input.value,
+                "verified": self.verified_input.value,
+                "event_id": self.event_id_input.value,
+            }
+
+class EventEditor:
+
+    def __init__(self, event_id=None):
+        self.event_id = event_id or str(uuid.uuid4())
+        self.event = {}
+        self.invites = []
+        self.dialog = None
+
+    async def load(self):
+        if self.event_id is not None:
+            self.event = await api.get_event(self.event_id)
+
+    async def save(self, on_save=None):
```

Comment:
```suggestion
    async def save(self, on_save: typing.Callable | None = None):
```

Reply:
Is it really necessary to type an optional callback function..?